slsim/image_simulation.py

```diff
@@ -395,10 +424,187 @@ def point_source_image_with_variability(
+
+
+def image_plus_poisson_noise(image, exposure_time):
+    """Creates an image with possion noise.
+
+    :param image: an image
+    :param exposure_time: exposure time or exposure map for an image
+    :return: image with possion noise
+    """
+    image[image < 0] = 0
+    mean_photons = image * exposure_time
+    return np.random.poisson(mean_photons) / exposure_time
+
+
+def image_plus_poisson_noise_for_list_of_image(images, exposure_times):
+    """Creates an image with possion noise.
+
+    :param images: list of images
+    :param exposure_time: list of exposure times or exposure maps
+    :return: list of images with possion noise
+    """
+    list_of_noisy_images = [
+        image_plus_poisson_noise(data, expo_time)
+        for data, expo_time in zip(images, exposure_times)
+    ]
+    return list_of_noisy_images
+
+
+def lens_image(
+    lens_class,
+    band,
+    mag_zero_point,
+    delta_pix,
+    num_pix,
+    psf_kernel,
+    transform_pix2angle,
+    exposure_time=None,
+    t_obs=None,
+):
+    """Creates lens image on the basis of given information. It can simulate both static
+    lens image and variable lens image.
+
+    :param lens_class: Lens() object
+    :param band: imaging band
+    :param mag_zero_point: magnitude zero point for the exposure
+    :param delta_pix: pixel scale of image generated
+    :param num_pix: number of pixels per axis
+    :param psf_kernels: psf kernel for the exposures being.
+    :param transform_pix2angle: transformation matrix (2x2) of pixels into coordinate
+        displacements
+    :param exposure_time: exposure time for for the exposure. It could be single
+        exposure time or a exposure map.
+    :param t_obs: an observation time [day]. This is applicable only for variable
+        source. In case of point source, if we do not provide t_obs, considers no
+        variability in the lens.
+    :return: lens image
+    """
+    deflector_source = sharp_image(
+        lens_class=lens_class,
+        band=band,
+        mag_zero_point=mag_zero_point,
+        delta_pix=delta_pix,
+        num_pix=num_pix,
+    )
+    convolved_deflector_source = convolved_image(
+        image=deflector_source, psf_kernel=psf_kernel
+    )
+    if t_obs is None:
+        image_ps = point_source_image_without_variability(
+            lens_class=lens_class,
+            band=band,
+            mag_zero_point=mag_zero_point,
+            delta_pix=delta_pix,
+            num_pix=num_pix,
+            psf_kernel=psf_kernel,
+            transform_pix2angle=transform_pix2angle,
+        )
+    else:
+        image_ps = point_source_image_at_time(
+            lens_class=lens_class,
+            band=band,
+            mag_zero_point=mag_zero_point,
+            delta_pix=delta_pix,
+            num_pix=num_pix,
+            psf_kernel=psf_kernel,
+            transform_pix2angle=transform_pix2angle,
+            time=t_obs,
+        )
+    image = convolved_deflector_source + sum(image_ps)
```

Comment:
are you sure this is doing the right thing? What is this sum over?

Reply:
Image_ps contains list of all the individual lensed images of a source. So, adding them gives us a all lensed images in one frame.